### PR TITLE
Add package detector for Kotlin files

### PR DESCRIPTION
**Pull request by 1c3t3a**
Changed Package Detectors functionality in a way that the Interface just wants a Regex Pattern matching the package name. Also implemented the KotlinPackageDetector.

### CODE REVIEW

File: src/test/java/edu/hm/hafner/analysis/KotlinPackageDetectorTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Tests the class {@link KotlinPackageDetector}.
+ *
+ * @author Bastian Kersting
+ */
+class KotlinPackageDetectorTest extends ResourceTest {
+    @ParameterizedTest(name = "{index} => file={0}, expected package={1}")
+    @CsvSource({
+            "MavenKotlinTest.txt, edu.hm.kersting",
+            "complicated-package-declaration-Kotlin.txt, edu.hm.kersting",
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/KotlinPackageDetectorTest.java:26]: Seems you are using a Windows machine, the test fails on Unix:
 
```suggestion
            "complicated-package-declaration-kotlin.txt, edu.hm.kersting",
```

Comment (1c3t3a) [reply src/test/java/edu/hm/hafner/analysis/KotlinPackageDetectorTest.java]: Fixed with ae1a39978dcc6a80d2951d0a012bb89a1d96e976

**Comment by uhafner** [comment]
There are 2 warnings left: https://ci.jenkins.io/job/Plugins/job/analysis-model/job/PR-476/5/pmd/new/

Otherwise all fine 👍

**Comment by 1c3t3a** [comment]
> There are 2 warnings left: https://ci.jenkins.io/job/Plugins/job/analysis-model/job/PR-476/5/pmd/new/
> 
> Otherwise all fine 👍

Alright! The PMD Warnings are now fixed in b404862a3fb60ea57745a87f3feb44ee69849503

**Comment by uhafner** [comment]
Thanks!

**Pull request merged**